docs(onpremise): Document adding custom CA roots (#369)
getsentry/develop: src/docs/self-hosted/custom-ca-roots.mdx

---
title: "Self-Hosted Custom CA Roots"
---

> Starting with Sentry `21.7.0`, if you need to have Sentry access services which do not have TLS certificates from publicly trusted CA roots, it's now possible to easily add them to the containers. Just add the certificates to the `certificates` folder inside the root of your Sentry install and restart the containers.

Reviewer: ```suggestion
Starting with Sentry `21.8.0`, if you need to have Sentry access services which do not have TLS certificates from publicly trusted CA roots, it's now possible to easily add them to the containers. Just add the certificates to the `certificates` folder inside the root of your Sentry install and restart the containers.
```

> Starting with Sentry `21.8.0`, if you need to have Sentry access services which do not have TLS certificates from publicly trusted CA roots, it's now possible to easily add them to the containers. Just add the certificates to the `certificates` folder inside the root of your Sentry install and restart the containers.Your custom CA roots will be used in addition to the publicly trusted CA roots.

Reviewer 2: ```suggestion
Starting with Sentry `21.8.0`, if you need to have Sentry access services which do not have TLS certificates from publicly trusted CA roots, it's now possible to easily add them to the containers. Just add the certificates to the `certificates` folder inside the root of your Sentry install and restart the containers. Your custom CA roots will be used in addition to the publicly trusted CA roots.
```

<Alert title="Note" level="info">

>   While you can run <a href="https://manpages.debian.org/buster/ca-certificates/update-ca-certificates.8.en.html"><code>update-ca-certificates</code></a> in each container, that will update the system's root bundle on disk, but does nothing for any copies in memory. Restarting the container will update the bundle and make sure it's used.

Reviewer 2: ```suggestion
  While you can run <a href="https://manpages.debian.org/buster/ca-certificates/update-ca-certificates.8.en.html"><code>update-ca-certificates</code></a> in each container, that will update the system's root bundle on disk, but does nothing for any copies in memory. Restarting the container will update the bundle and make sure it is used.
```

</Alert>

The container's logs will have the output from `update-ca-certificates` right at the start if there is a problem with a given certificate.

## Dependencies with bundled roots

> Some dependencies have opted to bundle their own CA roots and ingore the system CA roots. Where known, they have been configured to use the system roots. If something seems to ignore the system roots, make an issue so it can be tracked down and fixed.

Reviewer: ```suggestion
Some dependencies have opted to bundle their own CA roots and ignore the system CA roots. Where known, they have been configured to use the system roots. If something seems to ignore the system roots, make an issue so it can be tracked down and fixed.
```

### Overridden bundled roots

- Python

>   - request

Reviewer: ```suggestion
  - requests
```

> Some dependencies have opted to bundle their own CA roots and ignore the system CA roots. Where known, they have been configured to use the system roots. If something seems to ignore the system roots, [make an issue](https://github.com/getsentry/onpremise/issues/new?template=bug.yml) so it can be tracked down and fixed.

Reviewer 2: ```suggestion
Some dependencies have opted to bundle their own CA roots and ignore the system CA roots. Where known, they have been configured to use the system roots. If something seems to ignore the system roots, [create an issue](https://github.com/getsentry/onpremise/issues/new?template=bug.yml) so it can be tracked down and fixed.
```

---
title: "Self-Hosted Custom CA Roots"
---

Starting with Sentry `21.7.0`, if you need to have Sentry access services which do not have TLS certificates from publicly trusted CA roots, it's now possible to easily add them to the containers. Just add the certificates to the `certificates` folder inside the root of your Sentry install and restart the containers.

> 
> 

Reviewer 2: ```suggestion

```
